docs/how-to/contribute.md

```diff
@@ -0,0 +1,87 @@
+# How to contribute
+
+## Overview
+
+This document explains the processes and practices recommended for contributing enhancements to the Jenkins agent k8s operator.
+
+- Generally, before developing enhancements to this charm, you should consider [opening an issue
+  ](https://github.com/canonical/jenkins-agent-k8s-operator/issues) explaining your use case.
+- If you would like to chat with us about your use-cases or proposed implementation, you can reach
+  us at [Canonical Matrix public channel](https://app.element.io/#/room/#charmhub-charmdev:ubuntu.com)
+  or [Discourse](https://discourse.charmhub.io/).
+- Familiarising yourself with the [Charmed Operator Framework](https://juju.is/docs/sdk) library
+  will help you a lot when working on new features or bug fixes.
+- All enhancements require review before being merged. Code review typically examines
+  - code quality
+  - test coverage
+  - user experience for Juju operators of this charm.
+- Please help us out in ensuring easy to review branches by rebasing your pull request branch onto the `main` branch. This also avoids merge commits and creates a linear Git commit history.
+- Please generate src documentation for every commit. See the section below for more details.
+
+## Developing
+
+The code for this charm can be downloaded as follows:
+
+```
+git clone https://github.com/canonical/jenkins-agent-k8s-operator
+```
+
+You can use the environments created by `tox` for development:
+
+```shell
+tox --notest -e unit
+source .tox/unit/bin/activate
+```
+
+### Testing
+
+Note that the [Jenkins agent k8s](jenkins_agent_k8s_rock/rockcraft.yaml) needs to be built and pushed to microk8s for the tests to run. It should be tagged as `localhost:32000/jenkins-agent-k8s:latest` so that Kubernetes knows how to pull them from the microk8s repository. Note that the microk8s registry needs to be enabled using `microk8s enable registry`. More details regarding the OCI images below. The following commands can then be used to run the tests:
```

Comment:
Out of the scope of this PR but I think having `tox -e build` may help here :) https://github.com/canonical/jenkins-k8s-operator/pull/102 just FYI